src/main/java/com/seattleacademy/team20/SkillController.java

```diff
@@ -1,20 +1,119 @@
+import java.util.List;
 import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
 @Controller
 public class SkillController {
 	private static final Logger logger = LoggerFactory.getLogger(SkillController.class);
+	@Autowired
+	private JdbcTemplate jdbcTemplate;
 
+	//↑MySQLと接続するため
 	@RequestMapping(value = "/skillUpload", method = RequestMethod.GET)
-    public String dbAccessSample(Locale locale, Model model) {
-        logger.info("Welcome home! The client locale is {}.", locale);
-        return "skillUpload";
-    }
-}
\ No newline at end of file
+	public String upload(Locale locale, Model model) throws IOException {
+		logger.info("Welcome home! The client locale is {}.", locale);
+		initialize();
+		List<Skill> skills = selectSkills();
+		return "skillUpload";
+	}
+
+	//Listの宣言
+	public List<Skill> selectSkills() {
+		final String sql = "select * from skill";
+		return jdbcTemplate.query(sql, new RowMapper<Skill>() {
+			public Skill mapRow(ResultSet rs, int rowNum) throws SQLException {
+				return new Skill(rs.getString("name"), rs.getString("category"), rs.getInt("score"));
+			}
+		});
+	}
+
+	private FirebaseApp app;
+
+	//SDK初期化
+	public void initialize() throws IOException {
+		FileInputStream refreshToken = new FileInputStream(
+				"/Users/mihohiroshima/seattle-data/developportfolio-firebase-adminsdk-t4ypq-1c896f558e.json");
+		FirebaseOptions options = new FirebaseOptions.Builder()
+				.setCredentials(GoogleCredentials.fromStream(refreshToken))
+				.setDatabaseUrl("https://developportfolio.firebaseio.com/")
+				.build();
+		app = FirebaseApp.initializeApp(options, "other");
+	}
+
+	public void uploadSkill(List<Skill> skills) {
+		final FirebaseDatabase database = FirebaseDatabase.getInstance(app);
+		DatabaseReference ref = database.getReference("skillCategories");
+
+		List<Map<String, Object>> dataList = new ArrayList<Map<String, Object>>();
+		Map<String, Object> map;
+		Map<String, List<Skill>> skillMap = skills.stream().collect(Collectors.groupingBy(Skill::getCategory));
+		for (Map.Entry<String, List<Skill>> entry : skillMap.entrySet()) {
+			map = new HashMap<>();
+			map.put("category", entry.getKey());
+			map.put("category", entry.getValue());
```

Comment:
```suggestion
			map.put("skills", entry.getValue());
```
とかでなくて大丈夫？実際に確認できてる？